client/viame-web-common/components/AttributeInput.vue

```diff
@@ -19,58 +24,146 @@ export default Vue.extend({
       type: Array,
       default: () => [],
     },
+    focus: {
+      type: Boolean,
+      default: false,
+    },
   },
+  setup(props, { emit }) {
+    const tempVal = ref(null as null | boolean | number | string);
+    const inputBoxRef = ref(undefined as undefined | HTMLInputElement);
+    const boolOpts = [
+      { text: '', value: undefined },
+      { text: 'true', value: true },
+      { text: 'false', value: false },
+    ];
+    tempVal.value = props.value;
+    onMounted(() => {
+      if (props.focus && inputBoxRef.value) {
+        inputBoxRef.value.focus();
+      }
+    });
 
-  computed: {
-    foo() { return this.value; },
-  },
-
-  methods: {
-    change(newval: string | boolean | number | undefined): void {
-      const { name } = this;
-      let value;
-      switch (newval) {
-        case '':
-        case undefined:
-          value = undefined;
+    function blurType(e: KeyboardEvent) {
+      (e.target as HTMLInputElement).blur();
+    }
+    function onFocus() {
+      if (props.values && props.values.length) {
+        tempVal.value = null;
+      }
+    }
+    function onInputKeyEvent(e: KeyboardEvent) {
+      switch (e.code) {
+        case 'Escape':
+        case 'Enter':
+          blurType(e);
           break;
         default:
-          value = newval;
+          break;
       }
-      this.$emit('change', { name, value });
-    },
+    }
+
+    function change(event: InputEvent): void {
+      const target = event.target as HTMLInputElement;
+      const { name } = props;
+      if (target && target.value) {
+        const newval = target.value;
+        let value;
+        switch (newval) {
+          case '':
+          case ' ':
+          case undefined:
+            value = undefined;
+            break;
+          default:
+            value = newval;
+        }
+        emit('change', { name, value });
+      } else {
+        emit('change', { name, value: undefined });
+      }
+    }
+    return {
+      inputBoxRef,
+      tempVal,
+      boolOpts,
+      blurType,
+      onFocus,
+      onInputKeyEvent,
+      change,
+    };
   },
 });
 </script>
 
 <template>
   <div>
-    <v-combobox
+    <datalist
+      v-if="datatype === 'text' && values && values.length"
+      :id="`optionsList_${_uid}`"
```

Comment:
I'm just adding a comment as a note to make sure that I'm filtering out creating two attributes with the same name.  You can do it technically because the server references stuff by ID but for user confusion and any possible references I do by name I'm going to limit that.  I think that was the reason I initially added the error response to the Editor, just want to confirm that I did.